modules/apps/asset/asset-list-web/src/main/java/com/liferay/asset/list/web/internal/portlet/action/UpdateVariationsPriorityMVCActionCommand.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.asset.list.web.internal.portlet.action;
+
+import com.liferay.asset.list.constants.AssetListPortletKeys;
+import com.liferay.asset.list.model.AssetListEntrySegmentsEntryRel;
+import com.liferay.asset.list.service.AssetListEntrySegmentsEntryRelLocalService;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCActionCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCActionCommand;
+import com.liferay.portal.kernel.util.ParamUtil;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Yurena Cabrera
+ */
+@Component(
+	immediate = true,
+	property = {
+		"javax.portlet.name=" + AssetListPortletKeys.ASSET_LIST,
+		"mvc.command.name=/asset_list/update_variations_priority"
+	},
+	service = MVCActionCommand.class
+)
+public class UpdateVariationsPriorityMVCActionCommand
```

Comment:
Do this inside a transactional command.

Reply:
I saw you implemented the `updateVariationsPriority` at the service level but did not use it here. If you use it here you can avoid using a transactional command.

Reply:
You're absolutely right! Already refactored to use the method in the service instead